src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1525,10 +1626,14 @@ internal static uint GetRefEscape(BoundExpression expr, uint scopeOfTheContainin
                     return Binder.ExternalScope;
 
                 case BoundKind.RefValueOperator:
-                    // The undocumented __refvalue(tr, T) expression results in a variable of type T.
-                    // it is a readwrite variable, but could refer to unknown local data
-                    // we will constrain it to the current scope
-                    return scopeOfTheContainingExpression;
+                    // The undocumented __refvalue(tr, T) expression results in an lvalue of type T.
```

Comment:
__refvalue is used to "unwrap" the reference contained by TypedReference. Historically TypedReference ignores scoping, but cannot be returned from a method or passed by reference. Therefore it makes sense to assign the __refvalue Binder.TopLevelScope to not make it wider or narrower than effective scope of originating TypedReference value.